core/models.py

```diff
@@ -439,15 +467,51 @@ def save(self, *args, **kwargs):
         if self.end_date and timezone.is_naive(self.end_date):
             self.end_date = timezone.make_aware(self.end_date)
 
+        if self.review_status == ReviewStatus.Approved:
+            self.is_public = True
+        if self.review_status == ReviewStatus.Rejected:
+            self.is_public = False
+
+        h = Hackathon.objects.filter(
+            name=self.name,
+            end_date__year=self.end_date.year,
+            end_date__month=self.end_date.month,
+        )
+        if len(h) > 0 and h[0].id != self.id:
+            raise ValidationError(
+                f"An object with the name '{self.name}' already exists."
+            )
         super().save(*args, **kwargs)
 
 
+class Vote(models.Model):
+    type_vote = models.BooleanField(default=True)
+    hackathon_id = models.OneToOneField(
+        Hackathon, on_delete=models.CASCADE, related_name="votes"
+    )
+    from_hacker = models.ManyToManyField(Hacker, related_name="hackathon_votes")
+    time_created = models.DateTimeField(auto_now_add=True)
+    time_updated = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return f"Hacker(type_vote={self.type_vote}, hackathon_id={self.hackathon_id}, from_hacker={self.from_hacker}, time_created={self.time_created})"
+
+
 class CuratorRequest(models.Model):
     hackathon = models.ForeignKey(Hackathon, on_delete=models.CASCADE)
     team_name = models.CharField(max_length=255)
     team_description = models.TextField()
     reason = models.TextField()
     created_at = models.DateTimeField(auto_now_add=True)
+    created_by = models.ManyToManyField(Hacker, related_name="curator_requests")
```

Comment:
"curation_requests" is better

Reply:
what I should change to curation_requests I changed the     created_by = models.ManyToManyField(Hacker, related_name="curation_requests")